internal/db/schema/migrations/oss/postgres/22/01_session_valid_state.up.sql

```diff
@@ -0,0 +1,93 @@
+begin;
+
+-- Remove invalid (states with no prior end time) rows
+delete from session_state where previous_end_time is null and state != 'pending';
+
+-- In the case of dupes that are otherwise valid, identify first valid session
+create temp table session_state_delete
+as select session_id, min(start_time) as st_t from session_state ou
+   where (select count(*) from session_state inr
+          where inr.session_id = ou.session_id and inr.state = ou.state) > 1
+   group by session_id;
+
+-- Remove all session rows after first valid session
+DELETE FROM session_state t1
+    using session_state_delete t2 where t1.session_id =t2.session_id and t1.start_time > t2.st_t;
+
+drop table session_state_delete;
+
+-- Close all open session_connections
+update session_connection
+set closed_reason='canceled' where closed_reason is null;
+
+update session
+set termination_reason='canceled' where termination_reason is null;
+
+-- Migration
```

Comment:
I think this comment is trying to denote the end of the data cleanup steps and the start of the table modifications. If that is the case I would recommend expanding on this comment, since everything in this file can be thought of as a "migration". Alternatively, you could split these two steps into two migration files.